### PR TITLE
Class Variable / Constant Annotations

### PR DESCRIPTION
**Pull request by Jason2605**
### Well detailed description of the change :

This adds a new `.variableAnnotations` to classes that exposes any attached annotations to either class variables or class constants. Works exactly the same way as the other annotations where a dictionary is generated.

Resolves: #623 

#

### Type of change:

- [x] New feature

#

### Housekeeping

- [x] Tests have been updated to reflect the changes done within this PR (if applicable).

- [x] Documentation has been updated to reflect the changes done within this PR (if applicable).

#

### CODE REVIEW

**Comment by Jason2605** [comment]
@briandowns This should be working now - not sure I'm happy with `variableAnnotations` when it is also used for constants (contradicting a bit), do you have an idea for a better name?

**Comment by briandowns** [comment]
All of it looks great and I can't wait to finish the 2 modules I have in mind but as for the method name, I think I agree. How about `fieldAnnotations`?

**Comment by Jason2605** [comment]
Thanks man 😄 Yeah I like it, will change it to that

File: docs/docs/classes.md [hunk 1]
```diff
@@ -649,7 +649,7 @@ anotherTestObj.isInstance(Test); // true
 
 ## Annotations
 
-Annotations are metadata that are applied to classes and methods that by themselves have no impact. They, however, can provide user defined changes at runtime.
+Annotations are metadata that are applied to classes, methods and class variables that by themselves have no impact. They, however, can provide user defined changes at runtime.
```
Comment (briandowns) [inline docs/docs/classes.md:652]:
```suggestion
Annotations are metadata that are applied to classes, methods and class variables and constants that by themselves have no impact. They, however, can provide user defined changes at runtime.
```

**Pull request merged**